### PR TITLE
CDSR-1224: Main Declaration

### PR DESCRIPTION
**Pull request by steve-edmans**
Note main declaration and only include bank details with the main MRN

### CODE REVIEW

File: app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala [hunk 1]
```diff
@@ -625,12 +625,12 @@ object DefaultClaimTransformerService {
             MRNNumber = Some(mrn.value),
             acceptanceDate = Some(acceptanceDate),
             declarantReferenceNumber = displayDeclaration.displayResponseDetail.declarantReferenceNumber,
-            mainDeclarationReference = Some(true),
+            mainDeclarationReference = Some(completeClaim.movementReferenceNumber == mrn),
```
Comment (aneksamun) [inline app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala:628]: Does it complain on using double equals instead triple?
== vs ===
Normally you should see a warnings?

Comment (steve-edmans) [reply app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala:628]: No it didn't, I ran all of the tests and they worked.

Comment (steve-edmans) [reply app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala:628]: It turns out that === does not work for MRNs

Comment (aneksamun) [reply app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala:628]: @steve-edmans 
Please can you add
```
  implicit val eq: Eq[MRN] = Eq.fromUniversalEquals
```
in 

```
final case class MRN(value: String)

object MRN {

  def parse(value: String): Option[MRN] =
    if (isValid(value)) Some(new MRN(value)) else None

  def isValid(in: String): Boolean = {
    val regex = """\d{2}[a-zA-Z]{2}\w{13}\d""".r
    in match {
      case regex(_*) => true
      case _         => false
    }
  }

  implicit val binder: PathBindable[MRN] = new PathBindable[MRN] {
    def bind(key: String, value: String): Either[String, MRN] =
      parse(value).toRight(s"Value: $value cannot be parsed as MRN")

    def unbind(key: String, value: MRN): String = value.value

  }

  implicit val format: Format[MRN] = SimpleStringFormat(MRN(_), _.value)

// HERE
  implicit val eq: Eq[MRN] = ....
}

```

Comment (aneksamun) [reply app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala:628]: It will shut up linker

Comment (steve-edmans) [reply app/uk/gov/hmrc/cdsreimbursementclaim/services/ClaimTransformerService.scala]: I have made a simpler, more explicit change.

**Review by aneksamun — APPROVED** [review]

**Pull request merged**